backend/src/main/java/de/bund/digitalservice/ris/norms/application/service/VerkuendungsImportService.java

```diff
@@ -47,4 +88,200 @@ public UUID storeNormendokumentationspaket(Query query) throws IOException {
+    }
+
+    if (norm.getRegelungstexte().isEmpty() && norm.getBekanntmachungen().isEmpty()) {
+      throw new NoRegelungstextOrBekanntmachungstextException();
+    }
+
+    if (loadNormPort.loadNorm(new LoadNormPort.Command(norm.getWorkEli())).isPresent()) {
+      throw new NormExistsAlreadyException(norm.getWorkEli().toString());
+    }
+
+    log.info(
+      "Verified new norm from import: {} with {} Dokumenten and {} Binary files",
+      norm.getManifestationEli(),
+      norm.getDokumente().size(),
+      norm.getBinaryFiles().size()
+    );
+
+    return norm;
+  }
+
+  private void validateFileIsZipArchive(Resource file) {
+    var mediaType = MediaTypeFactory
+      .getMediaType(file)
+      .orElseThrow(() -> new NotAZipFileException("Invalid filetype: " + file.getFilename()));
+
+    if (!mediaType.includes(MimeType.valueOf("application/zip"))) {
+      throw new NotAZipFileException("Invalid filetype: " + file.getFilename());
+    }
+  }
+
+  private Set<Dokument> findParseAndValidateDokumente(Map<String, byte[]> files) {
+    if (!files.containsKey(RECHTSETZUNGSDOKUMENT_FILENAME)) {
+      throw new MissingRechtsetzungsdokumentException();
+    }
+
+    NormManifestationEli normManifestationEli = null;
+
+    Map<String, Dokument> dokumente = new HashMap<>();
+    Queue<String> dokumenteToProcess = new LinkedList<>();
+    dokumenteToProcess.add(RECHTSETZUNGSDOKUMENT_FILENAME);
+
+    while (!dokumenteToProcess.isEmpty()) {
+      String dokumentName = dokumenteToProcess.poll();
+      log.debug("Processing Dokument {}", dokumentName);
+      var dokument = parseAndValidateDokument(dokumentName, files.get(dokumentName));
+      dokumente.put(dokumentName, dokument);
+
+      if (!dokument.getManifestationEli().getSubtype().equals(dokumentName.split("\\.")[0])) {
+        throw new MismatchBetweenFilenameAndSubtypeException(
+          dokumentName,
+          dokument.getManifestationEli()
+        );
+      }
+
+      if (normManifestationEli == null) {
+        normManifestationEli = dokument.getManifestationEli().asNormEli();
+      }
+
+      if (!dokument.getManifestationEli().asNormEli().equals(normManifestationEli)) {
+        throw new InconsistentEliException(
+          dokumentName,
+          normManifestationEli,
+          dokument.getManifestationEli().asNormEli()
+        );
+      }
+
+      log.debug(
+        "Parsed & validated Dokument {} now looking for referenced dokuments in it",
+        dokument.getManifestationEli()
+      );
+
+      dokumenteToProcess.addAll(
+        dokument
+          .getReferencedDokumenteNames()
+          .stream()
+          // check if we already know about the referenced dokument
+          .filter(Predicate.not(dokumente::containsKey))
+          .filter(Predicate.not(dokumenteToProcess::contains))
+          .filter(referencedDokumentName -> {
+            if (isXmlFile(referencedDokumentName)) {
+              return true;
+            }
+            log.debug(
```

Comment:
Not 100% sure but the referenced documents are only XML? But they can also be PDF or other formats right? Like referencing the binary files with the componentRef in the Rechtsetzungsdokument?

Reply:
they are all xml files, for pdf attachments they are (as far as i know) a offene-struktur-xml that then contains a reference to the pdf file

Reply:
But page 121 of the specification (1.7.2) shows an example like:
```
<akn:component>
<akn:documentRef href="eli/bund/bgbl-1/2002/s100/deu/externes-dokument-1.pdf" showAs="externes-dokument"/>
</akn:component>
```
With the explanation:
_Ein rechtsetzungsdokument kann beliebig viele Teildokumente umfassen. Dies können auch Doku-
mente sein, die nicht im LegalDocML.de-Format vorliegen (etwa eine als PDF-Dokument vorliegende
Stellungnahme des Normenkontrollrates). Auch Nicht-LegalDocML.de-Dokumente werden anhand der
Elemente teildokument und teildokumentVerweis eingebunden_

Only _rechtsverbindlicher Anlagen_ will be referenced within the regelungstext

Reply:
Oh good catch, but it luckily doesn't change things about the code, i'll add a test with such a documentRef